src/watson/backends.py

```diff
@@ -274,7 +279,8 @@ def do_filter(self, engine_slug, queryset, search_text):
                 "watson_searchentry.search_tsv @@ to_tsquery('{search_config}', %s)".format(
                     search_config = self.search_config
                 ),
-                "watson_searchentry.{ref_name} = {table_name}.{pk_name}".format(
+                "watson_searchentry.{ref_name}{ref_name_typecast} = {table_name}.{pk_name}".format(
```

Comment:
Interesting. I wonder if we couldn't make it more generic by casting `{table_name}.{pk_name}` to string if it's not an integer primary key. That would presumably account more more fields than a UUID.